api/src/main/java/org/pmiops/workbench/monitoring/StackdriverStatsExporterService.java

```diff
@@ -25,93 +23,41 @@
   private static final Logger logger =
       Logger.getLogger(StackdriverStatsExporterService.class.getName());
   private static final String STACKDRIVER_CUSTOM_METRICS_PREFIX = "custom.googleapis.com/";
-  private static final String MONITORED_RESOURCE_TYPE = "generic_node";
-  private static final String PROJECT_ID_LABEL = "project_id";
-  private static final String LOCATION_LABEL = "location";
-  private static final String NAMESPACE_LABEL = "namespace";
-  private static final String NODE_ID_LABEL = "node_id";
-  public static final String UNKNOWN_INSTANCE_PREFIX = "unknown-";
 
-  private boolean initialized;
   private Provider<WorkbenchConfig> workbenchConfigProvider;
-  private ModulesService modulesService;
+  private Provider<MonitoredResource> monitoredResourceProvider;
 
   public StackdriverStatsExporterService(
-      Provider<WorkbenchConfig> workbenchConfigProvider, ModulesService modulesService) {
+      @Qualifier(CacheSpringConfiguration.WORKBENCH_CONFIG_SINGLETON)
+          Provider<WorkbenchConfig> workbenchConfigProvider,
+      Provider<MonitoredResource> monitoredResourceProvider) {
+    this.monitoredResourceProvider = monitoredResourceProvider;
+    logger.warning("Making StackdriverStatsExporterService");
     this.workbenchConfigProvider = workbenchConfigProvider;
-    this.modulesService = modulesService;
-    this.initialized = false;
   }
 
   /**
    * Create and register the stats configuration on the stats exporter. This operation should only
    * happen once, so we have an isInitialized guard for that.
    */
   public void createAndRegister() {
-    if (!initialized) {
-      try {
-        final StackdriverStatsConfiguration configuration = makeStackdriverStatsConfiguration();
-        StackdriverStatsExporter.createAndRegister(configuration);
-        logger.info(
-            String.format(
-                "Configured StackDriver exports with configuration:\n%s",
-                configuration.toString()));
-        initialized = true;
-      } catch (IOException e) {
-        logger.log(Level.WARNING, "Failed to initialize global StackdriverStatsExporter.", e);
-      }
+    try {
+      final StackdriverStatsConfiguration configuration = makeStackdriverStatsConfiguration();
+      StackdriverStatsExporter.createAndRegister(configuration);
+      logger.info(
+          String.format(
+              "Configured StackDriver exports with configuration:\n%s", configuration.toString()));
+    } catch (IOException e) {
+      logger.log(Level.WARNING, "Failed to initialize global StackdriverStatsExporter.", e);
     }
   }
 
   @VisibleForTesting
   public StackdriverStatsConfiguration makeStackdriverStatsConfiguration() {
     return StackdriverStatsConfiguration.builder()
         .setMetricNamePrefix(STACKDRIVER_CUSTOM_METRICS_PREFIX)
-        .setProjectId(getProjectId())
-        .setMonitoredResource(makeMonitoredResource())
+        .setProjectId(workbenchConfigProvider.get().server.projectId)
```

Comment:
Seems like the projectId can be fetched from monitoredResourceProvider

Reply:
yep.

Reply:
One caveat is that not every `MonitoredResource` type will have a label for `PROJECT_ID`. If I make this change, that makes this code a bit less reusable, since this service has no business knowing that we're in a GCP project.